From 9fb69e216362228ec0c4a4ef9ef2b547e67452ce Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Tue, 23 Jan 2018 23:22:52 -0800
Subject: [PATCH 1/2] sepolicy: Allow init to modify system_blk_device

* Only allow this for userdebug_or_eng since we are doing this
  to allow init to toggle verity state.

Change-Id: I6d41df69ac1db7d7dac3394a599521727dc8b719
---
 public/domain.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/domain.te b/public/domain.te
index 8640baaa..287db9dd 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -481,7 +481,7 @@ neverallow { domain -init -recovery -vold } metadata_block_device:blk_file
   { append link rename write open read ioctl lock };
 
 # No domain other than recovery and update_engine can write to system partition(s).
-neverallow { domain -recovery -update_engine } system_block_device:blk_file write;
+neverallow { domain -recovery -update_engine userdebug_or_eng(`-init') } system_block_device:blk_file write;
 
 # No domains other than install_recovery or recovery can write to recovery.
 neverallow { domain -install_recovery -recovery } recovery_block_device:blk_file write;
-- 
2.17.0

